core/src/main/java/org/testcontainers/containers/DockerComposeFiles.java

```diff
@@ -0,0 +1,33 @@
+package org.testcontainers.containers;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DockerComposeFiles {
+
+    private List<ParsedDockerComposeFile> parsedComposeFiles;
+
+    public DockerComposeFiles(List<File> composeFiles) {
+        this.parsedComposeFiles = composeFiles.stream().map(ParsedDockerComposeFile::new).collect(Collectors.toList());
+    }
+
+    public Set<String> getDependencyImages() {
+        Set<String> dependencyImageNames = new HashSet<>();
+
+        Collection<Set<String>> serviceDependencyImageNames = parsedComposeFiles.stream()
+            .flatMap(it -> it.getDependencyImageNames().entrySet().stream())
```

Comment:
It's also hard to read this part because you have no idea what `dependencyImageNames` is and what does that `flatmap` do.

Reply:
Reimplemented this part without stream